src/server/common/model/section/section-model/section-model.js

```diff
@@ -74,10 +79,11 @@ export class SectionModel {
     }
 
     if (page instanceof ExitPage) {
-      const questionPages = this.questionPages
+      const questionPageAnswers = this.questionPageAnswers
       return {
         isValid: false,
-        firstInvalidPage: questionPages[questionPages.length - 1]
+        firstInvalidPage:
+          questionPageAnswers[questionPageAnswers.length - 1].page
```

Comment:
whilst journey doesnt support this, based on this code, it is possible `this.questionPageAnswers.length` will be zero - so doing `.length -1` without any checking could be problematic

Reply:
The only way I can imagine to get this error is to do an explicit cast on a page to a QuestionPage:

```javascript
    class InvalidTestSection extends SectionModel {
      firstPage = /** @type QuestionPage */ (new Page()) // not a question page, as required
    }
```

Otherwise, we get an error from typescript.

The reason this guarantees a non-empty array is: 
- _pages pushes the firstPage first
- questionPageAnswers only filters on `instanceof QuestionPage`

So long as firstPage is a QuestionPage, then we won't run into this issue.